lib/patch-series.ts

```diff
@@ -718,9 +740,42 @@ export class PatchSeries {
             if (footers.length > 0) {
                 footers.push(""); // empty line
             }
-            // split the range-diff and prefix with a space
-            footers.push(`Range-diff vs v${this.metadata.iteration - 1}:\n\n${
-                         this.rangeDiff.replace(/(^|\n(?!$))/g, "$1 ")}\n`);
+
+            let rangeDiff = "";
+
+            if (typeof this.options.versionDiff === "undefined" || this.options.versionDiff) {
+                const args = [ this.rangeDiff.previousRange ];
+
+                if (this.rangeDiff.currentRange) {
+                    args.push(this.rangeDiff.currentRange);
+                }
+
+                // please set rangeDiff empty if post analysis decides it is too big or messy
+                rangeDiff = await git(["range-diff", "--no-color", "--creation-factor=95", ...args],
+                    { workDir: this.project.workDir });
+            }
+
+            if (rangeDiff) {
+                // split the range-diff and prefix with a space
+                footers.push(`Range-diff vs v${this.metadata.iteration - 1}:\n\n${
+                            rangeDiff.replace(/(^|\n(?!$))/g, "$1 ")}\n`);
```

Comment:
It would probably be better to move that into the previous `if` block, and if the range-diff _is_ empty, offer a different message (e.g. `Range-diff vs v${this.metadata.iteration - 1} is empty`).

Reply:
Will the diff ever be empty without an error?  I have adjusted the code. The previous iteration allowed post-range-diff analysis to set rangeDiff to empty but everything can be re-adjusted if that ever happens.

Reply:
It could be empty if a force-push happened after `git commit --amend` without editing the commit message (the commit date would have changed, and therefore the commit hash, but nothing that the range-diff cares about).

Reply:
After `git commit --amend --no-edit`:

```
git range-diff --creation-factor=95 f939873..64efcea5 f939873..a3b9bf7
1:  64efcea = 1:  a3b9bf7 Versioning a submit
```
So it still tells us if the commits are the same.